components/director/internal/domain/application/resolver.go

```diff
@@ -238,6 +254,62 @@ func (r *Resolver) Application(ctx context.Context, id string) (*graphql.Applica
 	return r.appConverter.ToGraphQL(app), nil
 }
 
+// ApplicationForTenant could be executed only in a certificate flow. It fetches model.ApplicationTemplate by label scenarioassignment.SubaccountIDKey
+// with value equal to the consumer ID. Lists all tenant header scoped []model.Application and finds the one created by the previously fetched App Template
+func (r *Resolver) ApplicationForTenant(ctx context.Context) (*graphql.Application, error) {
+	consumerInfo, err := consumer.LoadFromContext(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	if !consumerInfo.Flow.IsCertFlow() {
```

Comment:
In the spirit of the previous comment regarding moving this whole logic to the `applications` query - a similar check should be used to guard this logic from executing in the `applications` query resolver handler. Just one modification needed - don't just rely on checking that the flow is cert one (`consumerInfo.Flow.IsCertFlow()`) - the best way to guard this logic and be sure that this is a provider-consumer flow (i.e. two tenants are part of the tenant context) is to actually base the `if` check on `consumerInfo.OnBehalfOf != nil` - only when OnBehalfOf is existing can we be sure that there are two tenants at play (provider & consumer).